mcs/class/System/System.Net/HttpListenerRequest.cs

```diff
@@ -213,7 +213,7 @@ internal void FinishInitialization ()
 			if (raw_uri != null)
 				host = raw_uri.Host;
 	
-			int colon = host.IndexOf (':');
+			int colon = host.LastIndexOf (':');
```

Comment:
This is incorrect and needs to be reworked.  You cannot parse IPv6 address with with a simple `IndexOf()`, it is much more complex than that and I would recommend reusing existing APIs such as for instance `UriParser` for this.

IPv6 does not use colons to separate host and port numbers, but square brackets such as for instance

    https://[2001:db8:85a3:8d3:1319:8a2e:370:7348]:443/

and there can also be a scoped literal appended to it

    fe80::1ff:fe23:4567:890a%eth2

See https://en.wikipedia.org/wiki/IPv6_address for details.